tests/test_fieldtypes.py

```diff
@@ -1132,5 +1132,79 @@ def test_command_failed() -> None:
         command(b"failed")
 
 
+@pytest.mark.parametrize(
+    "path_cls",
+    [
+        fieldtypes.posix_path,
+        fieldtypes.windows_path,
+        fieldtypes.path,
+    ],
+)
+def test_empty_path(path_cls) -> None:
+    # initialize with empty string
+    p1 = path_cls("")
+    assert p1 == ""
+    assert p1._empty_path
+    assert str(p1) == ""
+    assert p1 != path_cls(".")
+
```

Comment:
Your solution is somewhat similar to my original proposal, which was rejected because it failed the following test (which also fails here):

```
 # initialize with empty path
    p1 = path_cls(Path(""))
    assert p1 == ""
    assert p1._empty_path
    assert str(p1) == ""
    assert p1 != path_cls(".")
```

Reply:
According to my information this was part of the requirements. I see no other way to implement it.

Reply:
As long as this works with our own `TargetPath`, all should be well in the world.

Reply:
There was not much information regarding the requirements in the DIS ticket, other than that Record path fields do not support empty values as they would end up as the `.` dot value. This PR fixes that.

If `windows_path` and `posix_path` fields are intitialised using `TargetPath` variables, then we need to ensure that `TargetPath` also supports empty paths and then we can test if that works.